File: src/python/plotter.py
import matplotlib.pyplot as plt
import numpy as np

def filereader(filename):
    data = np.loadtxt(filename)
    return data

def plotter(x, fwd, bwd, crank, n):
    plt.figure()
    plt.plot(x, fwd, '-o', label="Forward Euler", markersize=7, markevery=5)
    #plt.plot(x, fwd, '-o', label="Forward Euler")
    plt.plot(x, bwd, label="Backward Euler")
    plt.plot(x, crank, label="Crank-Nicholson")
    plt.xlabel("x")
    plt.ylabel(r'$u(x, t)$')
    plt.title('$u(x, t)$ with $t_{max} = 0.1$, $\Delta x = 0.00001$ and $\Delta t = 0.000005$')
    plt.legend()
    plt.savefig("../../plots/t1_" + str(n) + ".pdf")


def main():
    n = 100
    path = "../../data/"
    bwd = path + "bwd" + str(n) + ".txt"
    fwd = path + "fwd" + str(n) + ".txt"
    crank = path + "crank" + str(n) + ".txt"
    db = filereader(bwd)
    df = filereader(fwd)
    dc = filereader(crank)
    x = np.linspace(0, 1, len(df))
    plotter(x, df, db, dc, n)


if __name__ == '__main__':
    main()

File: src/matlab/plotter.m
home
close all
clear all

%Accessing the 'data' folder on github
addpath('..\..\data')

%Loading stable solutions
forward = load('fwd100.txt');
backward = load('bwd100.txt');
crank = load('crank100.txt');

%%

dx = linspace(0,1,length(forward)); %Defining x-axis

%Plotting the data
figure(1)
hold on
graphf = plot(dx,forward); %Forward
graphb = plot(dx,backward); %Backward
graphcn = plot(dx,crank); %Crank-Nicolson
set(graphf,'LineWidth',2); %Thicker lines
set(graphb,'LineWidth',2); %Thicker lines
set(graphcn,'LineWidth',2); %Thicker lines
title('Different methods over x') %Title
xlabel('x') %Labels and legend
ylabel('u(x)')
legend('Euler Forward','Euler Backward','Crank-Nicolson')
